Create install-the-oc-command-line-tool.md (#171)
bcgov/platform-developer-docs: src/docs/openshift-projects-and-access/install-the-oc-command-line-tool.md

---
title: Install the oc command line tool

slug: install-the-oc-command-line-tool

description: Describes how to install the oc command line tool for mac and windows 

keywords: OpenShift, command line, oc, brew, cli

page_purpose: Describes the installation of the oc command line tool for users preparing to use OpenShift 

audience: technical lead, openshift 101 students, developers

author: Matt Spencer  

editor: Pilar Solares

content_owner: Faisal Hamood

sort_order: 4
---

# Install the oc command line tool
Last updated: **May 30, 2023**

Users can interact with OpenShift via the `oc` command line tool. This can be installed on your local machine, or accessed via the web console. It is important to keep your version of `oc` up to date to match the version of `oc` on the cluster. 

## On this page
* <a href="section-1">**Install oc on a mac**</a>  
* <a href="section-2">**Install oc on windows**</a>   
* <a href="section-3">**Use web terminal**</a>  
* <a href="section-4">**Related pages**</a>

<!-- ### End of On this page -->

>  <h2 id="section-1">Install oc on a mac</h2> 

Reviewer: Is there a reason for using html instead of just markdown?
```suggestion
## Install oc on a mac
```
You can link to sections in markdown like this
```
* [**Install oc on a mac**](#install-oc-on-a-mac)
```

Reviewer 2: @Pilargit12 please note the format that Steven suggests here for how you can link to a heading markdown in markdown.

2. Click on your name at the top-right corner of the screen, and choose: Copy login command

3. A new tab will open in your browser. You'll need to login again by clicking the button: **Developer Log In** 

4. Next, you'll be directed to a page with a link to **Display Token**. Please do not share your tokens capture them on screen recordings as these grant access to OpenShift on your behalf. Click the link to reveal your tokens

5. Copy the entire line of text under: Log in with this token

6. Paste this command into your terminal and run it 

If successful, you should see an indication of which cluster you've logged in to, along with the number of projects you have access to and the project you are currently working in.

### Update oc
To ensure compatibility as features change, please update your `oc` command line tool regularly. You can use `brew upgrade openshift-cli` to achieve this if you installed `oc` using [homebrew](https://brew.sh/) as described above. 

<!-- ### Video demo  I'll comment this out until the video is ready-->

<h2 id="section-3"> Use the web terminal</h2>

> If you can't install `oc` locally, you may wish to use the OpenShift web terminal for some purposes. More information is [available from Redhat](https://docs.openshift.com/container-platform/4.8/web_console/odc-about-web-terminal.html#odc-using-web-terminal_odc-about-web-terminal). 

Reviewer: Use the current version of the docs
```suggestion
If you can't install `oc` locally, you may wish to use the OpenShift web terminal for some purposes. More information is [available from Redhat](https://docs.openshift.com/container-platform/4.12/web_console/odc-using-web-terminal.html#odc-access-web-terminal_odc-using-web-terminal). 
```

Reviewer 2: Thanks Steven! I had  tried switching to the latest version of the docs, but I guess it didn't pick it up since the name has changed and my search at the time didn't find this new one.

<br> 
---

<h2 id="section-4"> Related pages</h2>

- [Homebrew](https://brew.sh/)
- [Login to OpensShift Web Console](/login-to-openshift) 

> - [OpenShift Web Terminal - Redhat](https://docs.openshift.com/container-platform/4.8/web_console/odc-about-web-terminal.html#odc-using-web-terminal_odc-about-web-terminal)

Reviewer: ```suggestion
- [OpenShift Web Terminal - Redhat](https://docs.openshift.com/container-platform/4.12/web_console/odc-using-web-terminal.html#odc-access-web-terminal_odc-using-web-terminal)
```

> 

Reviewer: Add a link for installing the cli by downloading from the web console
```suggestion
- [Installing the OpenShift CLI by using the web console](https://docs.openshift.com/container-platform/4.12/cli_reference/openshift_cli/getting-started-cli.html#cli-installing-cli-web-console_cli-developer-commands)

```

---
title: Install the oc command line tool

slug: install-the-oc-command-line-tool

description: Describes how to install the oc command line tool for mac and windows 

keywords: OpenShift, command line, oc, brew, cli

page_purpose: Describes the installation of the oc command line tool for users preparing to use OpenShift 

audience: technical lead, openshift 101 students, developers

author: Matt Spencer  

editor: Pilar Solares

content_owner: Faisal Hamood

sort_order: 4
---

# Install the oc command line tool
Last updated: **May 31, 2023**

Users can interact with OpenShift via the `oc` command line tool. This can be installed on your local machine, or accessed via the web console. It is important to keep your version of `oc` up to date to match the version of `oc` on the cluster. 

## On this page
* [**Install oc on a mac**](#install-oc-on-a-mac) 
* [**Install oc on windows**](#install-oc-on-windows)

> * [**Use web terminal**](#use-web-terminal)

Reviewer 2: ```suggestion
* [**Use web terminal**](#use-the-web-terminal)
```

2. Click on your name at the top-right corner of the screen, and choose: Copy login command

3. A new tab will open in your browser. You'll need to login again by clicking the button: **Developer Log In** 

4. Next, you'll be directed to a page with a link to **Display Token**. Please do not share your tokens capture them on screen recordings as these grant access to OpenShift on your behalf. Click the link to reveal your tokens

5. Copy the entire line of text under: Log in with this token

6. Paste this command into your terminal and run it. If successful, you should see an indication of which cluster you've logged in to, along with the number of projects you have access to and the project you are currently working in.

### Update oc

To ensure compatibility as features change, please update your `oc` command line tool regularly. You can use `brew upgrade openshift-cli` to achieve this if you installed `oc` using [homebrew](https://brew.sh/) as described above. 

<!-- ### Video demo  I'll comment this out until the video is ready-->

##  Install oc on windows

### Install wsl 

> In the OpenShift 101 training, we suggest that new users install Windows Subsystem for Linux (WSL) first, and then install `oc` within `WSL` using [homebrew](https://brew.sh/). This is to maintain direct compatibility between the commands used across operating systems. 

Reviewer 2: ```suggestion
In the [OpenShift 101 training](/training-from-the-platform-services-team/), we suggest that new users install Windows Subsystem for Linux (WSL) first, and then install `oc` within `WSL` using [homebrew](https://brew.sh/). This is to maintain direct compatibility between the commands used across operating systems. 
```

5. Copy the entire line of text under: Log in with this token

6. Paste this command into your terminal and run it 

If successful, you should see an indication of which cluster you've logged in to, along with the number of projects you have access to and the project you are currently working in.

### Update oc
To ensure compatibility as features change, please update your `oc` command line tool regularly. You can use `brew upgrade openshift-cli` to achieve this if you installed `oc` using [homebrew](https://brew.sh/) as described above. 

<!-- ### Video demo  I'll comment this out until the video is ready-->

## Use the web terminal

If you can't install `oc` locally, you may wish to use the OpenShift web terminal for some purposes. More information is [available from Redhat](https://docs.openshift.com/container-platform/4.12/web_console/odc-using-web-terminal.html#odc-access-web-terminal_odc-using-web-terminal). 

<br>

---
## Related pages

- [Homebrew](https://brew.sh/)

> - [Login to OpensShift Web Console](/login-to-openshift) 

Reviewer 2: ```suggestion
- [Login to OpensShift Web Console](/login-to-openshift/) 
```
